bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/CommonInterface.java

```diff
@@ -181,16 +183,23 @@ default List<NAObject> updateReadings() {
     }
 
     default void commonInitialize(ScheduledExecutorService scheduler) {
-        setThingStatus(ThingStatus.UNKNOWN, null);
-        ModuleType moduleType = ModuleType.from(getThing().getThingTypeUID());
-        if (ModuleType.ACCOUNT.equals(moduleType.getBridge())) {
-            NAThingConfiguration config = getThing().getConfiguration().as(NAThingConfiguration.class);
-            getCapabilities().put(new RefreshCapability(this, scheduler, config.refreshInterval));
-        }
-        getCapabilities().values().forEach(cap -> cap.initialize());
-        CommonInterface bridgeHandler = getBridgeHandler();
-        if (bridgeHandler != null) {
-            bridgeHandler.expireData();
+        Bridge bridge = getBridge();
+        if (bridge == null || bridge.getHandler() == null) {
+            setThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED, null);
+        } else if (!ThingStatus.ONLINE.equals(bridge.getStatus())) {
+            setThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, null);
```

Comment:
Done by framework already?

Reply:
No when you need to override bridgeaStatusChanged to run specific init code.